### PR TITLE
Move project to JUnit 5 test platform

### LINKED ISSUE #2793: Replace deprecated DropwizardAppRule
As per Dropwizard documentation: 

> Deprecated since Dropwizard 2.0.0. Please migrate to JUnit 5 and io.dropwizard.testing.junit5.DropwizardAppExtension.

### PR DESCRIPTION
**Pull request by jose-west**
Since version 2.0.0 of Dropwizard, the `@ClassRule` annotations were deprecated. To support the new `@ExtendWith` notation, the test project has been moved to the JUnit5 platform. However this is not supported by the Powermock framework, so for unit tests using this the JUnit4 notations are used instead. Because the new platform does not support the old integration test configuration, these have been moved to a new subfolder under the test project, and are now run together with the other unit tests (which in turn simplifies the setup).

Fixes #2793 

- [ ] anet.yml needs change
- [ ] db needs migration
- [ ] documentation has changed

### CODE REVIEW

File: src/test/java/mil/dds/anet/test/integration/config/AnetTestConfiguration.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
   private static final HashMap<String, Object> config;
   private static final ObjectMapper yamlMapper = new ObjectMapper(new YAMLFactory());
   private static final File configFile =
-      new File("src/integration-test/resources/anet-integrationtest.yml");
+      new File("src/test/java/mil/dds/anet/test/integration/resources/anet-integrationtest.yml");
```
Comment (gjvoosten) [inline src/test/java/mil/dds/anet/test/integration/config/AnetTestConfiguration.java]: Shouldn't this file be in `src/test/resources` and loaded through the resource loader?

File: src/test/java/mil/dds/anet/test/integration/utils/FakeSmtpServer.java [hunk 2]
```diff
@@ -54,6 +54,11 @@ public FakeSmtpServer(SmtpConfiguration smtpConfig) throws Exception {
     // Not in config
     httpPort = System.getenv("ANET_SMTP_HTTP_PORT");
 
+    // We need the variable whcih depends on whether it is the test environment or not
+    if (httpPort == null) {
+      throw new Exception("'ANET_SMTP_HTTP_PORT' variable not found in system environment.");
+    }
+
```
Comment (gjvoosten) [inline src/test/java/mil/dds/anet/test/integration/utils/FakeSmtpServer.java]: Typo in the comment, plus probably better to `fail(…)` the test.

**Comment by jose-west** [comment]
> Please fix the LGTM alert: https://github.com/NCI-Agency/anet/pull/2836/checks?check_run_id=454683521

I do not mind fixing this, but it comes from the `candidate` branch. I will commit a fix.

File: src/main/java/mil/dds/anet/config/AnetConfiguration.java [hunk 3]
```diff
@@ -130,7 +130,7 @@ public void setServerUrl(String serverUrl) {
   }
 
   public Map<String, Object> getDictionary() {
-    return dictionary;
+    return new HashMap<>(dictionary);
```
Comment (VassilIordanov) [inline src/main/java/mil/dds/anet/config/AnetConfiguration.java]: can we instead do:
```
   public void setDictionary(Map<String, Object> dictionary) {
-    this.dictionary = dictionary;
+    this.dictionary = Collections.unmodifiableMap(dictionary);
   }
```

Comment (jose-west) [reply src/main/java/mil/dds/anet/config/AnetConfiguration.java]: Indeed, this way we do not have to create new copies each time the dictionary is requested.

**Pull request merged**